docs/rollout.md

```diff
@@ -0,0 +1,175 @@
+# Rollout Strategy in Fleet
+
+Fleet uses a rollout strategy to control how apps are deployed across clusters. This feature allows you to define the order and grouping of cluster deployments, enabling controlled rollouts, and safer updates. Fleet evaluates the **Ready** status of each `BundleDeployment` to determine when to proceed to the next group. For more information, refer to [Status Fields](./ref-status-fields.md).
+
+For initial deployments:
+
+* One cluster may be shown as **NotReady**.
+
+* Remaining clusters appear as **Pending**.
+
+For rollouts:
+
+* One cluster may be **NotReady**.
+
+* Remaining clusters are marked **OutOfSync** until the current one becomes ready
```

Comment:
We should clarify that this is about how the status messages of the GitRepos will look like when the deployments are started but not yet complete. So, for instance when the deployment is paused because it waits for the Bundles to become Ready.